### ISSUE TITLE
Filter in multiple properties of one resource

### ISSUE THREAD

**Issue by sunik** [cloudcreativity/laravel-json-api#59]
Hi!

I am newb and I was just wondering whether I do something the right way. 
I want to search for one value across multiple properties of one resource. Or rather I want to retrieve all records containing one value in multiple filters of one resource.

I believe this would be a reasonable representation of the URL design of what I want to achieve:
`api/v1/posts?filter[title,author,content]=somevalue`

Of course this does not work, so I created a new filter parameter called `all`:
 `api/v1/posts?filter[all]=somevalue`

I just added the `all` parameter to my allowed filters:

    protected $allowedFilteringParameters = [
        'id',
        'title',
        'author',
        'content',

        // For querying all of the above
        'all',
    ];

And then I used orWhere clauses on my query builder:

    // For querying all properties
    if ($filters->has('all')) {
        $builder->where('posts.title', 'like', '%' .  $filters->get('all') . '%')
                ->orWhere('posts.author', 'like', '%' .  $filters->get('all') . '%')
                ->orWhere('posts.content', 'like', '%' .  $filters->get('all') . '%');
    }

Now this works the way I want it to, but considering best practice and efficiency, Is this the right way to go about it?

**Comment by lindyhopchris** [comment]
Hi! Your solution looks sensible. Filtering logic is totally down to your domain logic has to how best to apply it for your application and database setup. That's why in the `Search` class I haven't written anything as to how to convert filter parameters to a query builder - because it's down to each application as to how best this is implemented.

In my applications I've done something similar where I've got a search term that can be applied against two database fields as an `or` query. I've done exactly as you have and had a filter that specifically means that, although I called it `search` rather than `all`. But same solution!
